From: Shannon Nelson <snelson@pensando.io>
Date: Wed, 10 Mar 2021 11:26:30 -0800
Subject: ionic: rebuild debugfs on qcq swap
Patch-mainline: v5.13-rc1
Git-commit: 55eda6bbe0c87c577376472f8bf381ef1c023303
References: jsc#SLE-16649

With a reconfigure of each queue is needed a rebuild of
the matching debugfs information.

Signed-off-by: Shannon Nelson <snelson@pensando.io>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/pensando/ionic/ionic_lif.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/pensando/ionic/ionic_lif.c
+++ b/drivers/net/ethernet/pensando/ionic/ionic_lif.c
@@ -2183,6 +2183,9 @@ static void ionic_swap_queues(struct ion
 	swap(a->cq_base,      b->cq_base);
 	swap(a->cq_base_pa,   b->cq_base_pa);
 	swap(a->cq_size,      b->cq_size);
+
+	ionic_debugfs_del_qcq(a);
+	ionic_debugfs_add_qcq(a->q.lif, a);
 }
 
 int ionic_reconfigure_queues(struct ionic_lif *lif,
